internal/index/indexoperations/index.go

```diff
@@ -72,6 +72,19 @@ func AddIndex(path, name, url string) error {
 	return errors.New("index already exists")
 }
 
+// DeleteIndex removes specified index name. If index does not exist, returns an error that can be tested by os.IsNotExist.
+func DeleteIndex(paths environment.Paths, name string) error {
+	if !IsValidIndexName(name) {
+		return errors.New("invalid index name")
+	}
+
+	dir := paths.IndexPath(name)
+	if _, err := os.Stat(dir); err != nil {
+		return err
+	}
+	return os.RemoveAll(dir)
```

Comment:
Should we be extra careful here and check that `dir` is not a symlink? E.g. `ln ~ ~/.krew/index/bad-bad-bad; krew index rm bad-bad-bad`

Reply:
Given we manage what's cloned directly under ~/.krew/index , this means user has done something malicious to themselves by messing w/ krew internals. Also, `git clone` produces these dirs –which won't be symlinks.

I prefer not to protect the user from their own mistakes. 

Furthermore `ln -s a b; rm -rf b;` doesn't actually remove `a/*` based on my experimentation. Correct?

I don't see any need for action here + I'd prefer to keep things simple (+not have to write tests for that scenario).

Reply:
On the command line, yes. `RemoveAll()` seems to behave identically, so yeah I was a bit to paranoid. Maybe I'm losing my sanity over corona :wink: